### PR TITLE
Add IT with 2 projects (test issues, measures)

### PR DESCRIPTION
**Pull request by andrei-epure-sonarsource**
First part of #3506

### CODE REVIEW

File: its/src/test/java/com/sonar/it/csharp/MultipleProjectsTest.java [hunk 1]
```diff
@@ -0,0 +1,424 @@
...
+    assertThat(getComponent(FIRST_PROJECT_FIRST_CLASS_FILE).getName()).isEqualTo("FirstClass.cs");
+    assertThat(getComponent(FIRST_PROJECT_SECOND_CLASS_FILE).getName()).isEqualTo("SecondClass.cs");
+    assertThat(getComponent(FIRST_PROJECT_ASSEMBLY_INFO_FILE).getName()).isEqualTo("AssemblyInfo.cs");
+
+    assertThat(getComponent(SECOND_PROJECT_DIRECTORY).getName()).isEqualTo("SecondProject");
+    assertThat(getComponent(SECOND_PROJECT_FIRST_CLASS_FILE).getName()).isEqualTo("FirstClass.cs");
+  }
+
+  @Test
+  public void barIssuesAreFound() {
```
Comment (andrei-epure-sonarsource) [inline its/src/test/java/com/sonar/it/csharp/MultipleProjectsTest.java:119]: sorry, the name of this class and `simpleClassIssueIsFound` are an artefact before a refactoring I will change it

File: its/src/test/java/com/sonar/it/csharp/MultipleProjectsTest.java [hunk 2]
```diff
@@ -0,0 +1,422 @@
...
+      .contains("24=1")
+      .contains("26=1")
+      .contains("27=1");
+
+    assertThat(secondProjectFirstClassNcloc.length()).isEqualTo(111); // No other line
+  }
+
+  /* Helper methods */
+
+  private Integer getProjectMeasureAsInt(String metricKey) {
```
Comment (andrei-epure-sonarsource) [inline its/src/test/java/com/sonar/it/csharp/MultipleProjectsTest.java:406]: the helper methods are duplicated with `MetricsTest` and could be factorized

**Comment by andrei-epure-sonarsource** [comment]
! Before merging, change the base of #3512 back to master

File: its/src/test/java/com/sonar/it/csharp/MultipleProjectsTest.java [hunk 3]
```diff
@@ -0,0 +1,422 @@
...
+  }
+
+  @Test
+  public void cognitiveComplexityAtFileLevel() {
+    assertThat(getFileMeasureAsInt(FIRST_PROJECT_FIRST_CLASS_FILE, "cognitive_complexity")).isEqualTo(2);
+    assertThat(getFileMeasureAsInt(FIRST_PROJECT_SECOND_CLASS_FILE, "cognitive_complexity")).isEqualTo(0);
+    assertThat(getFileMeasureAsInt(FIRST_PROJECT_ASSEMBLY_INFO_FILE, "cognitive_complexity")).isEqualTo(0);
+
+    assertThat(getFileMeasureAsInt(SECOND_PROJECT_FIRST_CLASS_FILE, "cognitive_complexity")).isEqualTo(2);
+  }
```
Comment (costin-zaharia-sonarsource) [inline its/src/test/java/com/sonar/it/csharp/MultipleProjectsTest.java:236]: This makes me think that we need to add tasks for updating the metrics when will add support for C#9. One example would be to count the Record types (at least as classes if not as a separate construct).

Btw, do we count structs or enums?

Comment (costin-zaharia-sonarsource) [reply its/src/test/java/com/sonar/it/csharp/MultipleProjectsTest.java:236]: I took a look and we do consider structs and interfaces classes from the metrics point of view: https://github.com/SonarSource/sonar-dotnet/blob/734df7968037387faade9353c7aedd1a77c010a0/sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/Common/MetricsTest.cs#L216.

Enums are **not** counted.

File: its/projects/MultipleProjects/FirstProject/FirstClass.cs [hunk 4]
```diff
@@ -0,0 +1,28 @@
+﻿using System;
+using System.Linq; // unused
+
+namespace FirstProject
+{
+    public class FirstClass
+    {
+        public int CoveredGet { get; set; } public int UncoveredProperty { get; set; } public int CoveredSet { get; set; }
+
+        public int CoveredGetOnSecondLine { get; set; }
```
[the remark this answers is not in the document]
Comment (andrei-epure-sonarsource) [reply its/projects/MultipleProjects/FirstProject/FirstClass.cs:10]: sorry, this is the test I used for Get / Set coverage when we did the sprint of adding branch coverage support.

File: its/projects/MultipleProjects/FirstProject/Properties/AssemblyInfo.cs [hunk 5]
```diff
@@ -0,0 +1,36 @@
+﻿using System.Reflection;
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
+
+// General Information about an assembly is controlled through the following
```
[the remark this answers is not in the document]
Comment (andrei-epure-sonarsource) [reply its/projects/MultipleProjects/FirstProject/Properties/AssemblyInfo.cs:5]: I'd leave the comment as we verify this as a metric

File: its/src/test/java/com/sonar/it/csharp/MultipleProjectsTest.java [hunk 6]
```diff
@@ -0,0 +1,422 @@
...
+    assertThat(getDirectoryMeasureAsInt(SECOND_PROJECT_DIRECTORY, "complexity")).isEqualTo(3);
+  }
+
+  @Test
+  public void complexityAtFileLevel() {
+    assertThat(getFileMeasureAsInt(FIRST_PROJECT_FIRST_CLASS_FILE, "complexity")).isEqualTo(13);
+    assertThat(getFileMeasureAsInt(FIRST_PROJECT_SECOND_CLASS_FILE, "complexity")).isEqualTo(1);
+    assertThat(getFileMeasureAsInt(FIRST_PROJECT_ASSEMBLY_INFO_FILE, "complexity")).isEqualTo(0);
+
+    assertThat(getFileMeasureAsInt(SECOND_PROJECT_FIRST_CLASS_FILE, "complexity")).isEqualTo(3);
```
[the remark this answers is not in the document]
Comment (costin-zaharia-sonarsource) [reply its/src/test/java/com/sonar/it/csharp/MultipleProjectsTest.java:213]: Related to this I've added:
https://github.com/SonarSource/sonar-dotnet/issues/3513
https://github.com/SonarSource/sonar-dotnet/issues/3516
https://github.com/SonarSource/sonar-dotnet/pull/3514

[the remark this answers is not in the document]
Comment (andrei-epure-sonarsource) [reply its/src/test/java/com/sonar/it/csharp/MultipleProjectsTest.java:213]: 👍  thanks for opening ##3516 and ##3513

File: its/src/test/java/com/sonar/it/csharp/MultipleProjectsTest.java [hunk 7]
```diff
@@ -0,0 +1,422 @@
...
+    assertThat(getProjectMeasureAsInt("cognitive_complexity")).isEqualTo(4);
+  }
+
+  @Test
+  public void cognitiveComplexityAtFileLevel() {
+    assertThat(getFileMeasureAsInt(FIRST_PROJECT_FIRST_CLASS_FILE, "cognitive_complexity")).isEqualTo(2);
+    assertThat(getFileMeasureAsInt(FIRST_PROJECT_SECOND_CLASS_FILE, "cognitive_complexity")).isEqualTo(0);
+    assertThat(getFileMeasureAsInt(FIRST_PROJECT_ASSEMBLY_INFO_FILE, "cognitive_complexity")).isEqualTo(0);
+
+    assertThat(getFileMeasureAsInt(SECOND_PROJECT_FIRST_CLASS_FILE, "cognitive_complexity")).isEqualTo(2);
```
[the remark this answers is not in the document]
Comment (andrei-epure-sonarsource) [reply its/src/test/java/com/sonar/it/csharp/MultipleProjectsTest.java:235]: 👍  thanks for opening ##3516 and ##3513

File: its/src/test/java/com/sonar/it/csharp/MultipleProjectsTest.java (hunk 3 above)
Comment (andrei-epure-sonarsource) [reply its/src/test/java/com/sonar/it/csharp/MultipleProjectsTest.java:236]: thanks!

File: its/src/test/java/com/sonar/it/csharp/MultiTargetAppTest.java [hunk 8]
```diff
@@ -20,22 +20,18 @@
...
 import org.junit.Test;
 import org.junit.rules.TemporaryFolder;
 import org.sonarqube.ws.Issues;
 
-import com.sonar.orchestrator.Orchestrator;
-
-import java.nio.file.Path;
-import java.util.List;
-
 import static com.sonar.it.csharp.Tests.ORCHESTRATOR;
```
Comment (costin-zaharia-sonarsource) [inline its/src/test/java/com/sonar/it/csharp/MultiTargetAppTest.java:35]: The import is no longer needed.

**Pull request merged**